sdk/test-utils/recorder/test/browser/recorder.spec.ts

```diff
@@ -0,0 +1,312 @@
+    const savedConsoleLogParams: any[] = [];
+    console.log = (...params: any) => {
+      savedConsoleLogParams.push(params);
+    };
+
+    // The recorder should start in the beforeEach call.
+    // We have to do this to emulate that.
+    const fakeThis: TestContextInterface = new TestContext(this.test! as TestContextTest, {
+      ...this.currentTest,
+      file: "test/recorder.browser.spec.ts",
+      // For this test, we don't care what's the content of the recorded function.
+      fn: getNoOpFunction()
+    });
+
+    const recorder = record(fakeThis, recorderEnvSetup);
+
+    // Restoring the XHR, otherwise we can't test this.
+    window.XMLHttpRequest = originalXHR;
+
+    const response = await helloWorldRequest();
+
+    // This test's request reached the server and received the expected response.
+    expect(response).to.equal(expectedHttpResponse);
+
+    // Cleaning everything before we continue verifying the results.
+    xhrMock.teardown();
+    recorder.stop();
+    console.log = originalConsoleLog;
+
+    // Here we confirm that the recorder generated an expected output on the console.logs.
+    // This output is used to generate the recording files in the filesystem, though here we're only
+    // checking what was that the recorded emitted to the standard output.
+    expect(savedConsoleLogParams[0][0]).to.equal(
+      // The recordings here are empty because we hijacked the XHR.
+      // See the playback test for an example of a properly constructed recording object.
+      // TODO: Find a way to capture the complete output.
+      JSON.stringify({
+        writeFile: true,
+        path:
+          "./recordings/browsers/the_recorders_public_api_on_a_browser/recording_should_record_a_simple_test.json",
+        content: {
+          recordings: [],
+          uniqueTestInfo: {
+            uniqueName: {},
+            newDate: {}
+          },
+          hash: MD5(getNoOpFunction().toString())
+        }
+      })
+    );
+  });
+
+  it("should playback a simple test", async function() {
+    // Setting up the playback mode.
+    // The PATH environment variable is not needed on playback.
+    // The recorder will assume that it is in playback mode by default.
+    // (window as any).__env__.TEST_MODE = "playback";
+
+    // This is to emulate what 'karma-json-preprocessor' does for us during the real scenarios.
+    (window as any).__json__ = {
+      ["recordings/browsers/the_recorders_public_api_on_a_browser/recording_should_playback_a_simple_test.json"]: {
+        recordings: [
+          {
+            method: "GET",
+            url: "/replaced",
+            response: expectedHttpResponse
+          }
+        ],
+        uniqueTestInfo: { uniqueName: {}, newDate: {} }
+      }
+    };
+
+    // The recorder should start in the beforeEach call.
+    // To emulate that behavior while keeping the test code as contained as possible,
+    // we're compensating with this.
+    const fakeThis: any = {
```

Comment:
this can be `TestContextInterface` yeah?

Reply:
oh thanks! I missed this